genesyscloud/resource_genesyscloud_journey_segment_test.go

```diff
@@ -56,30 +51,36 @@ func setupJourneySegment(t *testing.T, testCaseName string) {
 func cleanupJourneySegments(idPrefix string) {
 	journeyApi := platformclientv2.NewJourneyApiWithConfig(sdkConfig)
 
-	pageCount := 1 // Needed because of broken journey common paging
-	for pageNum := 1; pageNum <= pageCount; pageNum++ {
-		const pageSize = 100
+	segmentsToDelete := make([]platformclientv2.Journeysegment, 0)
+
+	// go through all segments to find those to delete
+	const pageSize = 200
+	for pageNum := 1; ; pageNum++ {
```

Comment:
Fetching all segments first then deleting the needed ones, as opposed to deleting and fetching.

I think with the previous code it could have bugged if trying to delete segments at the end of a page, then fetching the next page. It would have skipped a few segments.